web/packages/teleport/src/Discover/SelectResource/SelectResource.test.tsx

```diff
@@ -50,6 +52,24 @@ const makeResourceSpec = (
   );
 };
 
+/**
+ * If the user has resources, Connect My Computer is not prioritized when sorting resources.
+ */
+const onboardDiscoverWithResources: OnboardDiscover = {
+  notified: true,
+  hasResource: true,
+  hasVisited: true,
+};
+/**
+ * If the user does not have resources, Connect My Computer is prioritized as long as its resource
+ * spec is in the array and the user either has no preferences or prefers servers.
```

Comment:
*brain cramp* 😅

Reply:
I'll try to make it easier to understand, especially since here I'm just translating conditionals in code to a more verbose version in English.